controllers/placementapi_controller.go

```diff
@@ -750,79 +750,77 @@ func (r *PlacementAPIReconciler) initStatus(
 func (r *PlacementAPIReconciler) initConditions(
 	ctx context.Context, h *helper.Helper, instance *placementv1.PlacementAPI,
 ) error {
-	if instance.Status.Conditions == nil {
-		instance.Status.Conditions = condition.Conditions{}
-		// initialize conditions used later as Status=Unknown
-		cl := condition.CreateList(
-			condition.UnknownCondition(
-				condition.DBReadyCondition,
-				condition.InitReason,
-				condition.DBReadyInitMessage,
-			),
-			condition.UnknownCondition(
-				condition.DBSyncReadyCondition,
-				condition.InitReason,
-				condition.DBSyncReadyInitMessage,
-			),
-			condition.UnknownCondition(
-				condition.ExposeServiceReadyCondition,
-				condition.InitReason,
-				condition.ExposeServiceReadyInitMessage,
-			),
-			condition.UnknownCondition(
-				condition.InputReadyCondition,
-				condition.InitReason,
-				condition.InputReadyInitMessage,
-			),
-			condition.UnknownCondition(
-				condition.ServiceConfigReadyCondition,
-				condition.InitReason,
-				condition.ServiceConfigReadyInitMessage,
-			),
-			condition.UnknownCondition(
-				condition.DeploymentReadyCondition,
-				condition.InitReason,
-				condition.DeploymentReadyInitMessage,
-			),
-			// right now we have no dedicated KeystoneServiceReadyInitMessage and KeystoneEndpointReadyInitMessage
-			condition.UnknownCondition(
-				condition.KeystoneServiceReadyCondition,
-				condition.InitReason,
-				"Service registration not started",
-			),
-			condition.UnknownCondition(
-				condition.KeystoneEndpointReadyCondition,
-				condition.InitReason,
-				"KeystoneEndpoint not created",
-			),
-			condition.UnknownCondition(
-				condition.NetworkAttachmentsReadyCondition,
-				condition.InitReason,
-				condition.NetworkAttachmentsReadyInitMessage,
-			),
-			// service account, role, rolebinding conditions
-			condition.UnknownCondition(
-				condition.ServiceAccountReadyCondition,
-				condition.InitReason,
-				condition.ServiceAccountReadyInitMessage,
-			),
-			condition.UnknownCondition(
-				condition.RoleReadyCondition,
-				condition.InitReason,
-				condition.RoleReadyInitMessage,
-			),
-			condition.UnknownCondition(
-				condition.RoleBindingReadyCondition,
-				condition.InitReason,
-				condition.RoleBindingReadyInitMessage),
-			condition.UnknownCondition(
-				condition.TLSInputReadyCondition,
-				condition.InitReason,
-				condition.InputReadyInitMessage),
-		)
+	instance.Status.Conditions = condition.Conditions{}
```

Comment:
don't we want doing this conditionally, like cinder proposed https://github.com/openstack-k8s-operators/cinder-operator/pull/364/files?diff=split&w=1#diff-b57d545aef75db901bea7faa1e1e6fd8e93b88c0dc1638a8bb27258bf530c57aR169

Reply:
IMO this will not give us anything (exept that some conditions with status unknown will keep old LastTransitionTime) but we can change this too be consistent

Reply:
I agree it seems there is no need for nil checks

Reply:
So as far as I see `	instance.Status.Conditions.Init(&cl)` is smart and it updates existing conditions only if there was status change.

However `instance.Status.Conditions = condition.Conditions{}` blows away the previous list of conditions in the Status. So I think I see why cinder did this only once but calls Init unconditionally. 

With this PR a later reconcile:
* deletes the previous condition list
* creates a new condition list with unknown state (no last transition time)
* do the reconciliation and set condition states (conditions with updated state get a transition time)

Cinder's solution:
* keeps the previous condition list
* sets all condition state to Unknown in the list. (If the state wasn't unknown before then this updates the last transition time)
* do the reconciliation and set condition states accordingly (conditions with updated state get a transition time)

So basically both solutions make the last transition time of non Unknown conditions useless as both solution will reset that field at each reconciliation.

Also both solutions are noisy at the end as a later reconciliaton overwrites the transition time of the conditions in the API.

I will bring up loosing the last transition time issue on slack.